.github/workflows/continuous-integration.yml

```diff
@@ -214,3 +236,526 @@ jobs:
 
       - name: Audit dependencies
         run: yarn security-check
+
+  linting:
+    name: Linting (All)
+    if: ${{ github.ref == 'refs/heads/master' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Get Node version
+        id: get-node-version
+        run: echo ::set-output name=NODE_VERSION::$(cat .nvmrc)
+
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ steps.get-node-version.outputs.NODE_VERSION }}
+
+      - name: Cache dependencies
+        id: cache-dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/yarn
+            node_modules
+          key: ${{ steps.get-node-version.outputs.NODE_VERSION }}-${{ hashFiles('yarn.lock') }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --prefer-offline
+        env:
+          YARN_CACHE_FOLDER: ~/.cache/yarn
+
+      - name: Annotate ESLint results
+        run: yarn run eslint --ext .js --ext .jsx --format ./script/github-actions/eslint-annotation-format.js .
+
+      - name: Run Stylelint
+        run: yarn run stylelint verbose --output-file stylelint-report.json --formatter json src/**/*.scss
+
+      - name: Annotate Stylelint results
+        if: ${{ always() }}
+        run: node ./script/github-actions/stylelint-annotation-format.js
+
+  cypress-tests:
+    name: Cypress E2E Tests
+    runs-on: self-hosted
+    needs: build
+
+    strategy:
+      fail-fast: false
```

Comment:
Why would we not want `fail-fast` here?

Reply:
We want to let all tests complete, pass or fail, so that reporting can be comprehensive.